x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/import/errors.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { has } from 'lodash';
+
+export type RuleImportErrorType = 'conflict' | 'unknown';
+
+/**
+ * Generic interface representing a server-side failure during rule import.
+ * Used by utilities that import rules or related entities.
+ *
+ * NOTE that this does not inherit from Error
+ */
+export interface RuleImportErrorObject {
+  error: {
```

Comment:
Nit: Curious why it doesn't?

Reply:
I mainly added this because in the course of familiarizing myself with the import logic and `ImportRulesResponse`, I was surprised to find that `BulkError` also did not inherit from `Error`. I chose not to inherit from Error to keep things consistent.

As to _why_ it doesn't inherit from `Error`, I believe it's to allow for the addition of arbitrary fields on the object while not having to satisfy the `Error` interface with `name` and `message`.